graylog2-web-interface/src/views/components/searchbar/SearchButton.jsx

```diff
@@ -1,24 +1,60 @@
+// @flow strict
 import React from 'react';
 import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import type { StyledComponent } from 'styled-components';
 
 import { Button } from 'components/graylog';
 import { Spinner, Icon } from 'components/common';
+import type { ThemeInterface } from 'theme/index';
 
-const SearchButton = ({ running, disabled, glyph }) => (
-  <Button type="submit" bsStyle={running ? 'warning' : 'success'} disabled={disabled} className="pull-left search-button-execute">
-    {running ? <Spinner fixedWidth pulse text="" /> : <Icon name={glyph} /> }
-  </Button>
-);
+const DirtyButton: StyledComponent<{}, ThemeInterface, any> = styled(Button)`
+  position: relative;
+
+  ::after {
+    position: absolute;
+    content: '';
+    height: 16px;
+    width: 16px;
+    top: -5px;
+    right: -6px;
+    border-radius: 50%;
+    background-color: #ffc107;
+  }
+`;
+
+type Props = {
+  running: boolean,
+  disabled: boolean,
+  glyph: string,
+  dirty: boolean,
+};
+
+const SearchButton = ({ running, disabled, glyph, dirty }: Props) => {
+  const ButtonComponent = dirty ? DirtyButton : Button;
+  const title = dirty ? 'Perform search (changes were made after last search execution)' : 'Perform search';
+  return (
+    <ButtonComponent type="submit"
+                     bsStyle={running ? 'warning' : 'success'}
+                     disabled={disabled}
+                     title={title}
+                     className="pull-left search-button-execute">
+      {running ? <Spinner fixedWidth pulse text="" /> : <Icon name={glyph} />}
```

Comment:
Not related to this PR, but we are never defining the `running` prop. Are we not using the button loading state, because of to the global loading indicator?

Reply:
We never really used it, so I removed it.